### ISSUE TITLE
the update_class_critertion , this function comes from

### ISSUE THREAD

**Issue by yuheyuan** [layumi/AdaBoost_Seg#3]
```python
 def update_class_criterion(self, labels):
            weight = torch.FloatTensor(self.num_classes).zero_().cuda()
            weight += 1
            count = torch.FloatTensor(self.num_classes).zero_().cuda()
            often = torch.FloatTensor(self.num_classes).zero_().cuda()
            often += 1
            print(labels.shape)
            n, h, w = labels.shape
            for i in range(self.num_classes):
                count[i] = torch.sum(labels==i)
                if count[i] < 64*64*n: #small objective
                    weight[i] = self.max_value
            if self.often_balance:
                often[count == 0] = self.max_value

            self.often_weight = 0.9 * self.often_weight + 0.1 * often 
            self.class_weight = weight * self.often_weight
            print(self.class_weight)
            if self.focal:
                print('Using Focal loss')
                return FocalLoss(weight = self.class_weight, ignore_index=255, gamma = self.gamma)
            else:
                return nn.CrossEntropyLoss(weight = self.class_weight, ignore_index=255)
```
I find the update_class_criterion function in your trainer_ms.py.
I want to know who  created it in the first place because I want to change it somewhere, I need quote.
I think it's a very come class_criterion If you know, could you tell me.

**Comment by layumi** [comment]
Hi 

Yes. I follow the Dr. Yang Zou's work Self-training (ECCV18) to add the code. 
But I changed the hyper-parameter for PyTorch. 

I used this loss in both my ijcai20 and ijcv 21.  I just mentioned it in the implementation details.

**Comment by yuheyuan** [comment]
> Hi
> 
> Yes. I follow the Dr. Yang Zou's work Self-training (ECCV18) to add the code. But I changed the hyper-parameter for PyTorch.
> 
> I used this loss in both my ijcai20 and ijcv 21. I just mentioned it in the implementation details.

Oh,thank you.
